test/helpers/cmd.go

```diff
@@ -317,6 +352,70 @@ func (res *CmdRes) WaitUntilMatchRegexp(expr string, timeout time.Duration) erro
 		&TimeoutConfig{Timeout: timeout})
 }
 
+// WaitUntilMatchFilterLineTimeout applies the JSONPath 'filter' to each line of
+// `CmdRes.stdout` and waits until a line matches the 'expected' output.
+// If the 'timeout' is reached it will return an error.
+func (res *CmdRes) WaitUntilMatchFilterLineTimeout(filter, expected string, timeout time.Duration) error {
+	parser := jsonpath.New("").AllowMissingKeys(true)
+	err := parser.Parse(filter)
+	if err != nil {
+		return err
+	}
+
+	errChan := make(chan error, 1)
+	body := func() bool {
+		lines := res.ByLines()
+		for i, line := range lines {
+			if len(line) == 0 {
+				continue
+			}
+
+			var data interface{}
+			err := json.Unmarshal([]byte(line), &data)
+			if err != nil {
+				errChan <- fmt.Errorf("failed to parse line %d as JSON: %s", i, err)
+				return true
+			}
+
+			result := new(bytes.Buffer)
+			err = parser.Execute(result, data)
+			if err != nil {
+				errChan <- fmt.Errorf("failed to apply JSONPath to line %d: %s", i, err)
+				return true
+			}
+
+			if result.String() == expected {
+				return true
+			}
+		}
+
+		return false
+	}
+
+	err = WithTimeout(
```

Comment:
You might want to use the new `WaitUntilTrue()`

Reply:
I'm not sure what you are referring to? I did not find any references to `WaitUntilTrue`?